### PR TITLE
ES-3

### PR DESCRIPTION
**Pull request by bartguminiak**

### CODE REVIEW

File: ELSpace/Commons/Extensions/UIView_Embed.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import UIKit
+import Anchorage
+
+extension UIView {
+
+    static func viewWithEmbeddedContent(_ contentView: UIView) -> UIView {
+        let view = UIView(frame: .zero)
+        view.addSubview(contentView)
+        contentView.edgeAnchors == view.edgeAnchors + 8
```
Comment (Zaprogramiacz) [inline ELSpace/Commons/Extensions/UIView_Embed.swift]: If you want to hold this in UIView extension make it more generic and pass 8 value in parameter

File: ELSpace/Screens/Hub/Activity/ActivityView.swift [hunk 2]
```diff
@@ -54,5 +55,20 @@ private extension ActivityView {
...
 
+        static func scrollView() -> UIScrollView {
+            let scrollView = UIScrollView(frame: .zero)
+            scrollView.keyboardDismissMode = .interactive
+            scrollView.showsVerticalScrollIndicator = false
+            return scrollView
+        }
+
+        static func stackView() -> UIStackView {
+            let view = UIStackView(arrangedSubviews: [])
```
Comment (Zaprogramiacz) [inline ELSpace/Screens/Hub/Activity/ActivityView.swift]: Use initailizer with frame parameter

**Review by Zaprogramiacz — CHANGES_REQUESTED** [review]

File: ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        }
+    }
+
+    private let disposeBag = DisposeBag()
+
+    private var activityFormView: ActivityFormView! {
+        return view as? ActivityFormView
+    }
+
+    private func configureSubviews() {
```
Comment (Zaprogramiacz) [inline ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift:63]: Set initial state of views in `ActivityFormView`. Here you can set delegates

File: ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        activityFormView.hoursTextView.textField.autocorrectionType = .no
+        activityFormView.hoursTextView.textField.keyboardType = .decimalPad
+        activityFormView.hoursTextView.textField.delegate = self
+        activityFormView.commentTextView.textField.autocorrectionType = .no
+        activityFormView.commentTextView.textField.returnKeyType = .done
+        activityFormView.commentTextView.textField.delegate = self
+    }
+
+    private func setInitialState() {
+        activityFormView.dateTextView.separatorLine.backgroundColor = titleColorForState(false)
```
Comment (Zaprogramiacz) [inline ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift:70]: Look at previous comment

**Review by Zaprogramiacz — CHANGES_REQUESTED** [review]

File: ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift (hunk 4 above)
Comment (bartguminiak) [reply ELSpace/Screens/Hub/Activity/Form/ActivityFormViewController.swift:70]: I moved setup of static properties to view but dynamic ones left in viewController for keeping the logic in the same class

**Pull request merged**